### PR TITLE
Refactor to reduce lines of code in modules

### PR DESCRIPTION
**Pull request by hillstew**

### CODE REVIEW

File: lib/league_statistics.rb [hunk 1]
```diff
@@ -76,44 +69,41 @@ def defense_helper
   def best_defense
     goal_avg_by_team ||= defense_helper
     best_defense = goal_avg_by_team.key(goal_avg_by_team.values.min_by { |value| value})
-    team_name_helper(best_defense)
+    @teams[best_defense].team_name
```
Comment (lrs8810) [inline lib/league_statistics.rb:72]: Great job drying up the code!

**Review by lrs8810 — APPROVED** [review]

File: lib/league_statistics.rb [hunk 2]
```diff
@@ -122,16 +112,11 @@ def winningest_team
...
-    end[highest_team_id].team_name
+    @teams[highest_team_id].team_name
   end
 
   def best_fans
-    games_by_team = @game_teams.group_by do |game_team|
-      game_team.team_id
-    end
-
+    games_by_team = @game_teams.group_by(&:team_id)
```
Comment (grwthomps) [inline lib/league_statistics.rb:119]: Awesome refactoring here!

File: lib/league_statistics.rb [hunk 3]
```diff
@@ -144,21 +129,14 @@ def best_fans
...
       away = away_won_games.to_f / games_arr.length
-
       avg_by_team[team_id] = (home - away).round(2)
     end
-
     highest_team_id = avg_by_team.key(avg_by_team.values.max_by { |value| value})
-      @teams.select do |team_id, team|
-        highest_team_id == team_id
-    end[highest_team_id].team_name
+    @teams[highest_team_id].team_name
```
Comment (grwthomps) [inline lib/league_statistics.rb:135]: I really like this refactor!

File: lib/team_statistics.rb [hunk 4]
```diff
@@ -98,107 +73,82 @@ def rival(team_id)
...
     season_summary = {}
     games_by_type.each do |season_id, seasons_hash|
       season_summary[season_id] ||= {}
       seasons_hash.each do |season_type, games_arr|
-        if season_type == 'Postseason'
-          season_type = :postseason
-        elsif season_type == 'Regular Season'
-          season_type = :regular_season
-        end
+        season_type = :postseason if season_type == 'Postseason'
```
Comment (grwthomps) [inline lib/team_statistics.rb:136]: Simple and effective refactor, great job!

**Review by grwthomps — APPROVED** [review]
214 lines of code removed! Great job!

File: lib/league_statistics.rb (hunk 1 above)
Comment (tyladevon) [reply lib/league_statistics.rb:72]: This refactor is great!

**Review by tyladevon — APPROVED** [review]
Great refactor and dry-up of the code here team!

**Pull request merged**